Graphs/js/main-directed.js

```diff
@@ -827,3 +827,143 @@ function myPageRank() {
         animateNodes(ranks, 1000);
     }
 }
+
+function heldKarp() {
+    stopAnimation();
+    let root;
+    const [wm, id_index] = getWM(cy, false, true);
+
+    if(!isComplete()){
+        alert('Graph not complete. Please set the probability that two nodes are connected to 1.');
+        return;
+    } 
+
+    if (cy.elements(':selected').length >= 1) {
+        root = cy.elements(':selected')[0];
+    } else {
+        root = prompt('Please enter the id of the start node.\n', '1');
+        root = parseInt(root) - 1;
+    }
+
+    clearCyStyle();
+    
+    const nodes = getAllNodes(cy);
+
+    cy.elements(':grabbable').unselect();
+
+    heldKarpHelp(wm, root);
+}
+
+/**
+ *
+ * @param {number[][]} wm weight matrix
+ * @param {number} x starting point
+ */
+function heldKarpHelp(wm, x) {
+    let n = wm.length;
+    var memo = new Map();
```

Comment:
To set it as a global variable.

Reply:
The "var" modifier does not set a global variable. The difference between var and let is that let is block scoped while var is not. To set a global variable, you have to declare it in the top-level of your program. In your case, you don't need to set it as a global variable, because you already pass it into your helper function by reference.

See[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) for more info about "let".